src/components/patch-details/patch-details.tsx

```diff
@@ -17,16 +32,114 @@ interface PatchDetailsProps {
   assetPk: string;
   assetPatch?: Patch;
   assetArea?: Patch;
+  versionNumber?: number;
 }
 
-export const PatchDetails = ({ assetPk, assetPatch, assetArea }: PatchDetailsProps) => {
+export const buildUpdatePropertyPatchRequest = (
+  patchId: string,
+  areaId: string,
+): UpdatePropertyPatchRequest => ({
+  patchId,
+  areaId,
+});
+
+export const PatchDetails = ({
+  assetPk,
+  assetPatch,
+  assetArea,
+  versionNumber,
+}: PatchDetailsProps) => {
   const { heading } = locale.patchDetails;
 
   const { patchLabel, housingOfficerLabel, areaManagerLabel } = locale.patchDetails;
 
   const patchOrAreaDefined = assetPatch || assetArea;
   const housingOfficerName = assetPatch?.responsibleEntities[0]?.name;
   const areaManagerName = assetArea?.responsibleEntities[0]?.name;
+  const [newPatchName, setNewPatchName] = useState<string>();
+  const [isEditingPatchName, setIsEditingPatchName] = useState(false);
+  const [patchesAndAreas, setPatchesAndAreas] = useState<Patch[]>([]);
+
+  useEffect(() => {
+    getAllPatchesAndAreas().then((data) => {
+      setPatchesAndAreas(
+        data.filter(
+          (patchOrArea) =>
+            ![
+              "Hackney",
+              "CL Area",
+              "CP Area",
+              "HN1 Area",
+              "HN2 Area",
+              "SD Area",
+              "SH Area",
+              "SN Area",
+            ].includes(patchOrArea.name),
+        ),
+      );
+    });
+  }, []);
+
+  const updatePropertyPatchCall = (request: UpdatePropertyPatchRequest) => {
+    updatePropertyPatch(assetPk, request, versionNumber?.toString() ?? "")
+      .catch((err) => {
+        console.error(err);
+      })
+      .finally(() => {
+        window.location.reload();
```

Comment:
I believe this shouldn't need to refresh the page - should be a react state update just on the component